### PR TITLE
Add deprecated comments to rn

### PR DESCRIPTION
**Pull request by ostolero**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-822)

## Description
Added specific release notes comments for deprecated Integration, scripts, and playbook.
The comment will be added automatically when the `deprecate: true` flag is enabled in yml.

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:46]: what does custom mean in this name?

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:38]: lets return a set here of only the deprecated items

**Review by JudahSchwartz — COMMENTED** [review]
Very nice!
Will wait for unit tests for approval

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:46]: Haha not sure, will remove it.

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:44]:
```suggestion
    return { command.get('name') for command in commands if command.get('deprecated') }
```
Style choice =)

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:65]: do we care that these are yml objects? content_items seems a bit more clear to me

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:607]: if text isnt provided we'll have an extra blank line here. Lets add
if text:
    rn_desc += f'{text}\n'
rn_desc +=deprecate_rn

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:67]:
```suggestion
    if not old_yml.is_deprecated and new_yml.is_deprecated:
```

**Review by JudahSchwartz — COMMENTED** [review]
very nice! Great tests

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:47]: old_yml is CommentedMap, Changed to YAMLContentObject

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:67]: Not possible as old_yml is CommentedMap

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:51]: arent deprecate_line_with_replacement and deprecate_line_no_replacement lists?

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:39]: Please add a test

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:52]:
```suggestion
    return re.findall(DEPRECATED_REGEXES[0], description) or re.findall(DEPRECATED_REGEXES[1], description) or ''
```

Comment (JudahSchwartz) [reply demisto_sdk/commands/update_release_notes/update_rn.py:52]: I think this is a bit more clear, not a must obviously

**Review by JudahSchwartz — APPROVED** [review]
Great work! Please remember to add the test

File: demisto_sdk/commands/update_release_notes/update_rn.py
Comment (JudahSchwartz) [reply demisto_sdk/commands/update_release_notes/update_rn.py:52]: Nevermind

**Pull request merged**
